tools/rule_checker/main.go

```diff
@@ -19,29 +19,87 @@ package main
 import (
 	"flag"
 	"fmt"
-
-	"github.com/golang/glog"
+	"io"
+	"os"
 
 	"github.com/prometheus/prometheus/rules"
 )
 
-var ruleFile = flag.String("rule-file", "", "The path to the rule file to check.")
+var (
+	flagset  = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	ruleFile = flagset.String("rule-file", "", "The path to the rule file to check. (Deprecated)")
+)
+
+// checkFile reads from in or the file if in is nil. Sucessfully read rules
+// are printed to out.
```

Comment:
Hm, the function signature's meaning is a bit convoluted (actually opening the `filename`, but only if in is not `nil`, but also using it for the debug output in any case). How about moving the conditional branch & opening back to the caller of the function, and always pass in a valid `in` and `out`? Since this branch is only taken in one of this function's invocations, it will also not lead to any code duplication. It's specific to that one callsite.